File: week_01/python/chatbot.py
class ChatBot:
    
    def __init__(self, n):
        
        self.n = n
        
        self.chatBotMap = dict()
        
        self.chatBotMap["안녕"] = "어 안녕..."
        self.chatBotMap["잘가"] = "너도 조심히 가~!"
    
    def n_gram(self, document: str):
        
        return [
            document[i:i + self.n] for i in range(len(document)) \
            if len(document) >= i + self.n
        ]
    
    def answer(self, question: str):
        
        print("[나]:\t", question)
        
        nGram = self.n_gram(question)
        
        ans = "[봇]:\t무슨 말인지 못 알아 듣겠습니다."
        
        for w in nGram:
            if w in self.chatBotMap.keys():
                ans = "[봇]:\t%s" % self.chatBotMap.get(w)
        
        print(ans)
        
        return ans
    

File: week_01/java/src/ChatBot.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ChatBot {

    Map<String, String> chatBotMap;

    public void init() {

        chatBotMap = new HashMap<>();

        chatBotMap.put("안녕", "어 안녕...");
        chatBotMap.put("잘가", "너도 조심히 가~!");

    }

    public List<String> nGram(int n, String document) {

        List<String> rtnList = new ArrayList<>();

        for (int i = 0; i < document.length(); i++) {
            if (document.length() >= i + n) {
                rtnList.add(document.substring(i, i + n));
            }
        }

        return rtnList;

    }

    public String answer(String question) {

        System.out.println("[나]:\t" + question);

        String rtnStr = "";
        if (chatBotMap == null || chatBotMap.size() == 0)
            init();

        List<String> nGramList = nGram(2, question);
        // System.out.println(nGramList);

        String answerStr = "[봇]:\t무슨 말인지 못 알아 듣겠습니다.";
        for (String w : nGramList) {
            if (chatBotMap.containsKey(w)) {
                answerStr = "[봇]:\t" + chatBotMap.get(w);
            }
        }

        System.out.println(answerStr);

        return rtnStr;

    }
}
